source/common/http/http1/balsa_parser.cc

```diff
@@ -103,6 +104,17 @@ bool isUrlValid(absl::string_view url, bool is_connect) {
          std::all_of(path_query.begin(), path_query.end(), is_valid_path_query_char);
 }
 
+bool isVersionValid(absl::string_view version_input) {
+  // HTTP-version is defined at
+  // https://www.rfc-editor.org/rfc/rfc7230.html#section-2.6. HTTP/0.9 requests
```

Comment:
very optional: Can update to https://www.rfc-editor.org/rfc/rfc9112.html#section-2.3.

Reply:
Thank you for the suggestion.  Done!